packages/studio-web/src/app/upload/upload.component.ts

```diff
@@ -22,34 +30,14 @@ import {
   AlignmentProgress,
 } from "../soundswallower.service";
 import { TextFormatDialogComponent } from "../text-format-dialog/text-format-dialog.component";
-import { HttpErrorResponse } from "@angular/common/http";
 
 @Component({
   selector: "app-upload",
   templateUrl: "./upload.component.html",
   styleUrls: ["./upload.component.sass"],
 })
 export class UploadComponent implements OnInit {
-  langs$ = this.rasService.getLangs$().pipe(
-    map((langs: Array<SupportedLanguage>|HttpErrorResponse) =>
-    {
-    if (Array.isArray(langs)) {
-      return langs
-      .map((lang) => {
-        return { id: lang.code, name: lang.names["_"] };
```

Comment:
are we still sorting? or is this happening in the API?

Reply:
It happens in the client, not in the API.

Reply:
Unless that was changed recently.

Reply:
no, it wasn't changed. this change breaks our proper sorting so this should be reverted before accepting this PR

Reply:
<img width="197" alt="image" src="https://user-images.githubusercontent.com/14815080/219724584-27faa655-b1c1-457f-9100-f2672dd77e33.png">

Reply:
OK, then yeah, we can't remove client-side sorting. I didn't realize this PR did that.

Reply:
Hi, no, I don't want to *revert* this change because the old code was gross.  Instead I will just, like, sort the languages correctly.  Sorry, I didn't notice that we were doing that here.

Reply:
Thanks for fixing @dhdaines! Please avoid assessing code as “gross” in code reviews though.

Reply:
Ok, no problem! If it's your code I insulted, sorry about that!  I will use other descriptives like "confusing" or "needlessly complex" or "suspicious"